src/dbnode/persist/fs/cross_block_reader.go

```diff
@@ -0,0 +1,168 @@
+package fs
+
+import (
+	"bytes"
+	"container/heap"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/m3db/m3/src/x/checked"
+	xerrors "github.com/m3db/m3/src/x/errors"
+	"github.com/m3db/m3/src/x/ident"
+)
+
+var (
+	errReaderNotOrderedByIndex = errors.New("CrossBlockReader can only use DataFileSetReaders ordered by index")
+	_ CrossBlockReader = (*crossBlockReader)(nil)
```

Comment:
We could get the same compile time guarantees if `NewCrossBlockReader()` returned the interface type instead of the pointer type I think. That would be consistent w/ what I've seen in other places in the storage code. And then just cast to the pointer type in the unit tests.

Reply:
👍